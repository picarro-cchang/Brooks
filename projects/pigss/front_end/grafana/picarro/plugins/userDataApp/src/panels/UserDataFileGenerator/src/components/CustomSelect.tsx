import React, { Component } from 'react';
import Select, { components } from 'react-select';

const customStyles = {
  container: provided => ({
    ...provided,
    minWidth: '10%',
  }),
  menu: provided => ({
    ...provided,
    padding: 5,
    backgroundColor: '#262628',
  }),
  option: provided => ({
    ...provided,
    color: '#d8d9da',
    backgroundColor: '#262628',
  }),
  control: provided => ({
    ...provided,
    color: '#d8d9da',
    backgroundColor: '#09090b',
    border: '1px solid #262628',
  }),
  multiValue: provided => ({
    ...provided,
    color: '#d8d9da',
    fontSize: 14,
    backgroundColor: '#09090b',
  }),
  multiValueLabel: provided => ({
    ...provided,
    color: '#d8d9da',
    fontSize: 14,
    backgroundColor: '#09090b',
  }),
};

const Option = props => {
  return (
    <div style={{ display: 'flex' }}>
      <components.Option {...props}>
        <label>
          <input type="checkbox" checked={props.isSelected} onChange={e => null} />
          <span className="checkbox-label">
            {'   '}
            {props.value}
          </span>
        </label>
      </components.Option>
    </div>
  );
};

const ClearIndicator = props => {
  return null;
};

export interface Props {
  passedOptions: [];
  passedOnChange: (keys: any) => void;
  allOption: {
    label: string;
    value: string;
  };
  value: [];
}

export default class CustomSelect extends Component<Props> {
  constructor(props) {
    super(props);
  }
  render() {
    const { passedOptions, passedOnChange, value } = this.props;
    return (
      <Select
        placeholder="Select..."
        components={{ Option, ClearIndicator }}
        styles={customStyles}
        options={passedOptions}
        onChange={passedOnChange}
        value={value}
        // value={passedOptions.find((option: any) => option.value === 'key')}
        isMulti={true}
        closeMenuOnSelect={false}
        hideSelectedOptions={false}
        backspaceRemovesValue={true}
      />
    );
  }
}
