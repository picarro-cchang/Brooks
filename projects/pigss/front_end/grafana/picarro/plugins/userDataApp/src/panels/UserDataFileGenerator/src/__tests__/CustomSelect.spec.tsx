import React from 'react';
import CustomSelect, {Props} from './../components/CustomSelect';
import { shallow } from 'enzyme';

const defaultProps: Props = {
    passedOptions: [],
    passedOnChange: (x: any) => null,
    allOption: {
        value: 'All',
        label: 'All'
    },
    value: []
}
describe('<DataGeneratorPanel/>', () => {
  const wrapper = shallow(<CustomSelect {...defaultProps}  />);

  it('Create Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
