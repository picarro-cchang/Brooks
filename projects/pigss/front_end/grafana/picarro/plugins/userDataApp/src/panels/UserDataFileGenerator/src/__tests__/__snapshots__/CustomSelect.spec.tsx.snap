// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGeneratorPanel/> Create Snapshot 1`] = `
<StateManager
  allOption={
    Object {
      "label": "All",
      "value": "All",
    }
  }
  backspaceRemovesValue={true}
  classNamePrefix="list"
  closeMenuOnSelect={false}
  components={
    Object {
      "ClearIndicator": [Function],
      "MultiValue": [Function],
      "Option": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  hideSelectedOptions={false}
  isMulti={true}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "H2O",
        "value": "H2O",
      },
      Object {
        "label": "CO2",
        "value": "CO2",
      },
    ]
  }
  passedOnChange={[Function]}
  passedOptions={
    Array [
      Object {
        "label": "H2O",
        "value": "H2O",
      },
      Object {
        "label": "CO2",
        "value": "CO2",
      },
    ]
  }
  placeholder="Select..."
  styles={
    Object {
      "container": [Function],
      "control": [Function],
      "menu": [Function],
      "multiValue": [Function],
      "multiValueLabel": [Function],
      "option": [Function],
    }
  }
/>
`;
